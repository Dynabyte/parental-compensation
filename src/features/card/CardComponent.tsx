import React from "react";

interface Props {
  arrData: any[];
  classNamne?: string;
}

const CardComponent: React.FC<Props> = ({ arrData, classNamne }) => {
  return (
    <ul>
      {arrData.map((data, index) => (
        <li key={index} className={classNamne}>
          {data.text} {data.value}
        </li>
      ))}
    </ul>
  );
};

export default CardComponent;
