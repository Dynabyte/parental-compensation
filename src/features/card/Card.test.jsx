import CardComponent from "./CardComponent";
import CardrReducer from "./CardReducer";

describe("Testin CardComponent", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CardComponent arrData={[]} />);
  });

  it("Should render empty ul if arrData is empty", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Testin CardReducer", () => {
  it("should return initialState", () => {
    const initialState = [];

    expect(CardrReducer(initialState, {})).toEqual(initialState);
  });
});
