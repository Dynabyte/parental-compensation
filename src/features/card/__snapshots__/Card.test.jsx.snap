// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testin CardComponent Should render empty ul if arrData is empty 1`] = `<ul />`;
